### ISSUE TITLE
crash when removing semicolon in tetris example

### ISSUE THREAD

**Issue by Extrawurst** [buggins/dlangide#21]
When removing the semicolon after the lambda in the tetris example here:
![image](https://cloud.githubusercontent.com/assets/776816/6173064/6a8ba1da-b2e5-11e4-89e0-457b24a101f6.png)

this used to work IIRC, now an exception is thrown:
```
core.exception.RangeError@src\ddc\lexer\tokenizer.d(1798): Range violation
----------------
0x005025A8 in _d_arraybounds
0x0044A0AC in ddc.lexer.tokenizer.Token ddc.lexer.tokenizer.Tokenizer.processIdent() at E:\_docs\_github\mydlangide\src\ddc\lexer\tokenizer.d(1798)
0x0044C545 in ddc.lexer.tokenizer.Token ddc.lexer.tokenizer.Tokenizer.nextToken() at E:\_docs\_github\mydlangide\src\ddc\lexer\tokenizer.d(2734)
0x0046CDD7 in void dlangide.ui.dsourceedit.SimpleDSyntaxHighlighter.updateHighlight(immutable(dchar)[][], ubyte[][], int, int) at E:\_docs\_github\mydlangide\src\dlangide\ui\dsourceedit.d(635)
0x004A63C6 in void dlangui.core.editable.EditableContent.updateTokenProps(int, int) at C:\Users\Stephan\AppData\Roaming\dub\packages\dlangui-master\src\dlangui\core\editable.d(677)
0x004A7488 in void dlangui.core.editable.EditableContent.handleContentChange(dlangui.core.editable.EditOperation, ref dlangui.core.editable.TextRange, ref dlangui.core.editable.TextRange, Object) at C:\Users\Stephan\AppData\Roaming\dub\packages\dlangui-master\src\dlangui\core\editable.d(909)
0x004A8865 in bool dlangui.core.editable.EditableContent.performOperation(dlangui.core.editable.EditOperation, Object) at C:\Users\Stephan\AppData\Roaming\dub\packages\dlangui-master\src\dlangui\core\editable.d(1221)
0x00479CAE in bool dlangui.widgets.editors.EditWidgetBase.removeRangeText(dlangui.core.editable.TextRange) at C:\Users\Stephan\AppData\Roaming\dub\packages\dlangui-master\src\dlangui\widgets\editors.d(908)
0x0047A62C in bool dlangui.widgets.editors.EditWidgetBase.handleAction(const(dlangui.core.events.Action)) at C:\Users\Stephan\AppData\Roaming\dub\packages\dlangui-master\src\dlangui\widgets\editors.d(1067)
0x0049DB6C in bool dlangui.widgets.editors.EditBox.handleAction(const(dlangui.core.events.Action)) at C:\Users\Stephan\AppData\Roaming\dub\packages\dlangui-master\src\dlangui\widgets\editors.d(2091)
0x0046AD5C in bool dlangide.ui.dsourceedit.DSourceEdit.handleAction(const(dlangui.core.events.Action)) at E:\_docs\_github\mydlangide\src\dlangide\ui\dsourceedit.d(87)
0x004D63BD in bool dlangui.platforms.common.platform.Window.dispatchAction(const(dlangui.core.events.Action), dlangui.widgets.widget.Widget) at C:\Users\Stephan\AppData\Roaming\dub\packages\dlangui-master\src\dlangui\platforms\common\platform.d(672)
0x00487AC4 in bool dlangui.widgets.widget.Widget.dispatchAction(const(dlangui.core.events.Action)) at C:\Users\Stephan\AppData\Roaming\dub\packages\dlangui-master\src\dlangui\widgets\widget.d(1002)
0x00487DA9 in bool dlangui.widgets.widget.Widget.onKeyEvent(dlangui.core.events.KeyEvent) at C:\Users\Stephan\AppData\Roaming\dub\packages\dlangui-master\src\dlangui\widgets\widget.d(1053)
0x0047B91A in bool dlangui.widgets.editors.EditWidgetBase.onKeyEvent(dlangui.core.events.KeyEvent) at C:\Users\Stephan\AppData\Roaming\dub\packages\dlangui-master\src\dlangui\widgets\editors.d(1366)
0x004D5CFB in bool dlangui.platforms.common.platform.Window.dispatchKeyEvent(dlangui.core.events.KeyEvent) at C:\Users\Stephan\AppData\Roaming\dub\packages\dlangui-master\src\dlangui\platforms\common\platform.d(535)
0x004D9E67 in bool dlangui.platforms.windows.winapp.Win32Window.onKey(dlangui.core.events.KeyAction, uint, int, dchar, bool) at C:\Users\Stephan\AppData\Roaming\dub\packages\dlangui-master\src\dlangui\platforms\windows\winapp.d(688)
0x004DBDFE in WndProc at C:\Users\Stephan\AppData\Roaming\dub\packages\dlangui-master\src\dlangui\platforms\windows\winapp.d(1115)
0x7566791C in CallNextHookEx
0x75667B7A in CallNextHookEx
0x756697F4 in GetMessageW
0x75669859 in DispatchMessageW
0x004C42E9 in int dlangui.platforms.windows.winapp.Win32Platform.enterMessageLoop() at C:\Users\Stephan\AppData\Roaming\dub\packages\dlangui-master\src\dlangui\platforms\windows\winapp.d(743)
0x00451DC7 in UIAppMain at E:\_docs\_github\mydlangide\src\dlangide.d(62)
0x004A05E6 in int dlangui.platforms.windows.winapp.myWinMain(void*, void*, char*, int) at C:\Users\Stephan\AppData\Roaming\dub\packages\dlangui-master\src\dlangui\platforms\windows\winapp.d(1013)
0x004764B8 in DLANGUIWinMain at C:\Users\Stephan\AppData\Roaming\dub\packages\dlangui-master\src\dlangui\platforms\windows\winapp.d(856)
0x00451C9E in WinMain at C:\Users\Stephan\AppData\Roaming\dub\packages\dlangui-master\src\dlangui\platforms\common\platform.d(1311)
0x005478C2 in WinMainCRTStartup
0x751F919F in BaseThreadInitThunk
0x771C0BBB in RtlInitializeExceptionChain
0x771C0B91 in RtlInitializeExceptionChain
```

**Comment by Extrawurst** [comment]
well here is another one:
```
core.exception.RangeError@src\dlangide\ui\dsourceedit.d(653): Range violation
----------------
0x005025A8 in _d_arraybounds
0x0046CE98 in void dlangide.ui.dsourceedit.SimpleDSyntaxHighlighter.updateHighlight(immutable(dchar)[][], ubyte[][], int, int) at E:\_docs\_github\mydlangide\src\dlangide\ui\dsourceedit.d(653)
0x004A63C6 in void dlangui.core.editable.EditableContent.updateTokenProps(int, int) at C:\Users\Stephan\AppData\Roaming\dub\packages\dlangui-master\src\dlangui\core\editable.d(677)
0x004A7488 in void dlangui.core.editable.EditableContent.handleContentChange(dlangui.core.editable.EditOperation, ref dlangui.core.editable.TextRange, ref dlangui.core.editable.TextRange, Object) at C:\Users\Stephan\AppData\Roaming\dub\packages\dlangui-master\src\dlangui\core\editable.d(909)
0x004A8865 in bool dlangui.core.editable.EditableContent.performOperation(dlangui.core.editable.EditOperation, Object) at C:\Users\Stephan\AppData\Roaming\dub\packages\dlangui-master\src\dlangui\core\editable.d(1221)
0x0047B903 in bool dlangui.widgets.editors.EditWidgetBase.onKeyEvent(dlangui.core.events.KeyEvent) at C:\Users\Stephan\AppData\Roaming\dub\packages\dlangui-master\src\dlangui\widgets\editors.d(1361)
0x004D5CFB in bool dlangui.platforms.common.platform.Window.dispatchKeyEvent(dlangui.core.events.KeyEvent) at C:\Users\Stephan\AppData\Roaming\dub\packages\dlangui-master\src\dlangui\platforms\common\platform.d(535)
0x004D9E67 in bool dlangui.platforms.windows.winapp.Win32Window.onKey(dlangui.core.events.KeyAction, uint, int, dchar, bool) at C:\Users\Stephan\AppData\Roaming\dub\packages\dlangui-master\src\dlangui\platforms\windows\winapp.d(688)
0x004DBE9A in WndProc at C:\Users\Stephan\AppData\Roaming\dub\packages\dlangui-master\src\dlangui\platforms\windows\winapp.d(1130)
0x7566791C in CallNextHookEx
0x75667B7A in CallNextHookEx
0x756697F4 in GetMessageW
0x75669859 in DispatchMessageW
0x004C42E9 in int dlangui.platforms.windows.winapp.Win32Platform.enterMessageLoop() at C:\Users\Stephan\AppData\Roaming\dub\packages\dlangui-master\src\dlangui\platforms\windows\winapp.d(743)
0x00451DC7 in UIAppMain at E:\_docs\_github\mydlangide\src\dlangide.d(62)
0x004A05E6 in int dlangui.platforms.windows.winapp.myWinMain(void*, void*, char*, int) at C:\Users\Stephan\AppData\Roaming\dub\packages\dlangui-master\src\dlangui\platforms\windows\winapp.d(1013)
0x004764B8 in DLANGUIWinMain at C:\Users\Stephan\AppData\Roaming\dub\packages\dlangui-master\src\dlangui\platforms\windows\winapp.d(856)
0x00451C9E in WinMain at C:\Users\Stephan\AppData\Roaming\dub\packages\dlangui-master\src\dlangui\platforms\common\platform.d(1311)
0x005478C2 in WinMainCRTStartup
0x751F919F in BaseThreadInitThunk
0x771C0BBB in RtlInitializeExceptionChain
0x771C0B91 in RtlInitializeExceptionChain
```

this happens when removing (backspace) random characters at the end of the file that were randomly typed there before:
![image](https://cloud.githubusercontent.com/assets/776816/6173319/3acd8632-b2e7-11e4-9a92-a8700c2213f7.png)

clearly this syntax highlighter needs more error resistance, after closing the exception messagebox the whole IDE app closes...

**Comment by buggins** [comment]
let me check

**Comment by buggins** [comment]
Fixed.
Thank you!

**Comment by Extrawurst** [comment]
nice!
by the way you can write stuff like "fixes #21" in the commit message and github automatically links the commit and the issue for intuitive navigation
